src/v/kafka/server/handlers/configs/config_utils.h

```diff
@@ -184,25 +187,96 @@ ss::future<std::vector<R>> do_alter_topics_configuration(
 
 template<typename T, typename R>
 ss::future<std::vector<R>>
-do_alter_broker_configuartion(std::vector<T> resources) {
-    // for now we do not support altering any of brokers config, generate
-    // errors
+do_alter_broker_configuartion(request_context& ctx, std::vector<T> resources) {
     std::vector<R> responses;
     responses.reserve(resources.size());
-    std::transform(
-      resources.begin(),
-      resources.end(),
-      std::back_inserter(responses),
-      [](T& resource) {
-          return make_error_alter_config_resource_response<R>(
-            resource,
-            error_code::invalid_config,
-            fmt::format(
-              "changing '{}' broker property isn't currently supported",
-              resource.resource_name));
-      });
 
-    return ss::make_ready_future<std::vector<R>>(std::move(responses));
+    // If central config is disabled, we cannot set broker properties
+    if (!config::node().enable_central_config()) {
+        std::transform(
+          resources.begin(),
+          resources.end(),
+          std::back_inserter(responses),
+          [](T& resource) {
+              return make_error_alter_config_resource_response<R>(
+                resource,
+                error_code::invalid_config,
+                fmt::format(
+                  "changing '{}' broker property isn't currently supported",
+                  resource.resource_name));
+          });
+
+        co_return responses;
+    }
+
+    for (const auto& resource : resources) {
+        cluster::config_update_request req;
+        bool errored = false;
+        for (const auto& c : resource.configs) {
+            if constexpr (std::
+                            is_same_v<T, incremental_alter_configs_resource>) {
+                auto op = static_cast<config_resource_operation>(
+                  c.config_operation);
+                if (op == config_resource_operation::set) {
+                    req.upsert.push_back({c.name, c.value.value()});
+                } else if (op == config_resource_operation::remove) {
+                    req.remove.push_back(c.name);
+                } else {
+                    responses.push_back(
+                      make_error_alter_config_resource_response<R>(
+                        resource,
+                        error_code::invalid_config,
+                        fmt::format(
+                          "operation {} on broker properties isn't currently "
+                          "supported",
+                          op)));
+                    errored = true;
+                    continue;
+                }
+            } else {
```

Comment:
I'm wondering if the `else` block needs to be value dependent (and if it isn't already)? there is some case for the catch-all conditions of a constexpr if statement where the body of the condition is dropped in the instantiated version. but I cannot remember all the details such as if it only applies to a case like the static_assert workaround listed here https://en.cppreference.com/w/cpp/language/if.

maybe a test already handles checking this case.

Reply:
This conditional went away when removing the AlterConfigs path -- the function is no longer generic because it's always dealing with an incremental request.